test/test_torch.py

```diff
@@ -16626,6 +16630,65 @@ def test_nextafter(self, device, dtype):
         expected = np.nextafter(a.cpu().numpy(), b.cpu().numpy())
         self.assertEqual(actual, expected, atol=0, rtol=0)
 
+    @dtypesIfCUDA(torch.float16, torch.float32, torch.float64)
+    @dtypesIfCPU(torch.bfloat16, torch.float32, torch.float64)
+    @dtypes(torch.float32, torch.float64)
+    @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
+    def test_i0_range1(self, device, dtype):
+        a = torch.rand(100, device=device).to(dtype) * 13.25
+        actual = torch.i0(a)
+        if dtype is torch.bfloat16:
+            a = a.to(torch.float32)
+        expected = scipy.special.i0(a.cpu().numpy())
+        if dtype is torch.bfloat16 or dtype is torch.float16:
```

Comment:
why the float16 check here?

Reply:
Answering my own question: because of the upcasting that SciPy does. A comment would be helpful.